homeassistant/components/local_todo/store.py

```diff
@@ -0,0 +1,38 @@
+"""Local storage for the Local To-do integration."""
+
+import asyncio
+from pathlib import Path
+
+from homeassistant.core import HomeAssistant
+
+STORAGE_PATH = ".storage/{key}.ics"
+
+
+class LocalTodoListStore:
+    """Local storage for a single To-do list."""
+
+    def __init__(self, hass: HomeAssistant, path: Path) -> None:
+        """Initialize LocalTodoListStore."""
+        self._hass = hass
+        self._path = path
+        self._lock = asyncio.Lock()
+
+    async def async_load(self) -> str:
+        """Load the calendar from disk."""
+        async with self._lock:
+            return await self._hass.async_add_executor_job(self._load)
+
+    def _load(self) -> str:
+        """Load the calendar from disk."""
+        if not self._path.exists():
+            return ""
+        return self._path.read_text()
```

Comment:
If there's an OSError here, should we handle that and make the entity unavailable?

Reply:
Currently this is read once at startup, kept in memory, then persisted only when the service changes. It does not regularly reload from disk to look for updates. I think the spirit of this comment, however, can be addressed by the other discussion around checking for store loading errors a setup time, which has been updated to check for OSError with some tests.